litmus-portal/backend/subscriber/pkg/cluster/cluster.go

```diff
@@ -92,25 +123,62 @@ func ClusterOperations(clientData map[string]interface{}, kubeconfig *string) (*
 	// Decode YAML manifest into unstructured.Unstructured
 	obj := &unstructured.Unstructured{}
 	_, gvk, err := decUnstructured.Decode([]byte(yamlStr), nil, obj)
-		if err != nil {
-		return nil, fmt.Errorf("err: %v\n", err)
+	if err != nil {
+		return nil, err
 	}
 
 	// Find GVR
 	mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
 	if err != nil {
-		return nil, fmt.Errorf("err: %v\n", err)
+		return nil, err
 	}
 
 	// Obtain REST interface for the GVR
 	var dr dynamic.ResourceInterface
 	if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
 		// namespaced resources should specify the namespace
-		dr = dynamicClient.Resource(mapping.Resource).Namespace(obj.GetNamespace())
+		//dr = dynamicClient.Resource(mapping.Resource).Namespace(
+		dr = dynamicClient.Resource(mapping.Resource).Namespace("litmus")
 	} else {
 		// for cluster-wide resources
 		dr = dynamicClient.Resource(mapping.Resource)
 	}
 
-	return applyRequest(obj, requestType, dr)
+	if requestType == "create" {
+		response, err := dr.Create(s.Ctx, obj, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		log.Println("Resource successfully applied")
+		return response, nil
+	} else if requestType == "update" {
+		response, err := dr.Update(s.Ctx, obj, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		log.Println("Resource successfully applied")
+		return response, nil
+	} else if requestType == "delete" {
+		err := dr.Delete(s.Ctx, obj.GetName(), metav1.DeleteOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		log.Println("Resource successfully applied")
+		return &unstructured.Unstructured{}, nil
+	} else if requestType == "get" {
+		response, err := dr.Get(s.Ctx, obj.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		log.Println("Resource successfully applied")
+		return response, nil
+	} else {
+		return nil, fmt.Errorf("err: %v\n", "Invalid Request")
+	}
+
+	return nil, fmt.Errorf("Invalid request")
```

Comment:
These lines are moved from applyRequest() function because this function is used only by the clusteroperation() function. also, it will be difficult to understand if we separate these if/else condition